### ISSUE TITLE
'list' object has no attribute 'items'\r\n"

### ISSUE THREAD

**Issue by xiemeilong** [ansible/ansible-modules-core#4363]
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

docker
##### ANSIBLE VERSION

```
ansible 2.1.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### STEPS TO REPRODUCE

```

---
- hosts: all
  remote_user: root
  tasks:
  - name: pull run business web
    docker:
      name: business-web
      image: "xxxxxx:10150/yuanmai/business-web:{{version}}"
      insecure_registry: yes
      state: reloaded
      pull: always
      restart_policy: always
      restart_policy_retry: 3
      extra_hosts:
        consul-8500.service.consul: "{{consul}}"
      volumes:
      - /opt/web/file:/opt/web/file
      - /etc/localtime:/etc/localtime:ro
      - /etc/timezone:/etc/timezone:ro
      ports:
      - "8085:8080"
  - name: pull run billing web
    docker:
      name: billing-web
      image: "xxxxxxx:10150/yuanmai/billing-web:{{version}}"
      insecure_registry: yes
      state: reloaded
      restart_policy: always
      restart_policy_retry: 3
      extra_hosts:
        consul-8500.service.consul: "{{consul}}"
      volumes:
      - /opt/web/file:/opt/web/file
      - /etc/localtime:/etc/localtime:ro
      - /etc/timezone:/etc/timezone:ro
      pull: always
      ports:
      - "8084:8080"
  - name: pull run billing pay service
    docker:
      name: billing-pay
      image: "xxxx:10150/yuanmai/billing-pay:{{version}}"
      insecure_registry: yes
      state: reloaded
      restart_policy: always
      restart_policy_retry: 3
      extra_hosts:
        consul-8500.service.consul: "{{consul}}"
      volumes:
      - /etc/localtime:/etc/localtime:ro
      - /etc/timezone:/etc/timezone:ro
      pull: always
      ports:
      - "8086:9090"
```

```
ansible-playbook deploy.yml  --extra-vars "version=Release-1.0.17"  -vvv  -f 4 -i production
```

##### EXPECTED RESULTS

```
PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
ok: [test02]

TASK [pull run business web] ***************************************************
changed: [test02]

TASK [pull run billing web] ****************************************************
changed: [test02]

TASK [pull run billing pay service] ********************************************
changed: [test02]

PLAY RECAP *********************************************************************
test02                     : ok=4    changed=3    unreachable=0    failed=0   
```
##### ACTUAL RESULTS

```
TASK [setup] *******************************************************************
<58.215.178.243> ESTABLISH SSH CONNECTION FOR USER: root
<58.215.178.243> SSH: EXEC ssh -C -q -o ControlMaster=auto -o ControlPersist=60s -o Port=6322 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/xiemeilong/.ansible/cp/ansible-ssh-%h-%p-%r 58.215.178.243 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1470289137.72-245841536783431 `" && echo ansible-tmp-1470289137.72-245841536783431="` echo $HOME/.ansible/tmp/ansible-tmp-1470289137.72-245841536783431 `" ) && sleep 0'"'"''
<58.215.178.243> PUT /tmp/tmpUExJBz TO /root/.ansible/tmp/ansible-tmp-1470289137.72-245841536783431/setup
<58.215.178.243> SSH: EXEC sftp -b - -C -o ControlMaster=auto -o ControlPersist=60s -o Port=6322 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/xiemeilong/.ansible/cp/ansible-ssh-%h-%p-%r '[58.215.178.243]'
<58.215.178.243> ESTABLISH SSH CONNECTION FOR USER: root
<58.215.178.243> SSH: EXEC ssh -C -q -o ControlMaster=auto -o ControlPersist=60s -o Port=6322 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/xiemeilong/.ansible/cp/ansible-ssh-%h-%p-%r -tt 58.215.178.243 '/bin/sh -c '"'"'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1470289137.72-245841536783431/setup; rm -rf "/root/.ansible/tmp/ansible-tmp-1470289137.72-245841536783431/" > /dev/null 2>&1 && sleep 0'"'"''
ok: [web01]

TASK [pull run business web] ***************************************************
task path: /home/xiemeilong/Projects/LogisticsBilling/bin/ansible/deploy.yml:5
<58.215.178.243> ESTABLISH SSH CONNECTION FOR USER: root
<58.215.178.243> SSH: EXEC ssh -C -q -o ControlMaster=auto -o ControlPersist=60s -o Port=6322 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/xiemeilong/.ansible/cp/ansible-ssh-%h-%p-%r 58.215.178.243 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1470289138.64-252795182131668 `" && echo ansible-tmp-1470289138.64-252795182131668="` echo $HOME/.ansible/tmp/ansible-tmp-1470289138.64-252795182131668 `" ) && sleep 0'"'"''
<58.215.178.243> PUT /tmp/tmpFu07WR TO /root/.ansible/tmp/ansible-tmp-1470289138.64-252795182131668/docker
<58.215.178.243> SSH: EXEC sftp -b - -C -o ControlMaster=auto -o ControlPersist=60s -o Port=6322 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/xiemeilong/.ansible/cp/ansible-ssh-%h-%p-%r '[58.215.178.243]'
<58.215.178.243> ESTABLISH SSH CONNECTION FOR USER: root
<58.215.178.243> SSH: EXEC ssh -C -q -o ControlMaster=auto -o ControlPersist=60s -o Port=6322 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/xiemeilong/.ansible/cp/ansible-ssh-%h-%p-%r -tt 58.215.178.243 '/bin/sh -c '"'"'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1470289138.64-252795182131668/docker; rm -rf "/root/.ansible/tmp/ansible-tmp-1470289138.64-252795182131668/" > /dev/null 2>&1 && sleep 0'"'"''
fatal: [web01]: FAILED! =>
 {"changed": false, "failed": true, "invocation": {"module_name": "docker"}, "module_stderr": "", "module_stdout": 
"Traceback (most recent call last):
File \"/tmp/ansible_8msQQL/ansible_module_docker.py\", line 1975, in <module>
    main()\r\n  File \"/tmp/ansible_8msQQL/ansible_module_docker.py\", line 1945, in main    reloaded(manager, containers, count, name)
  File \"/tmp/ansible_8msQQL/ansible_module_docker.py\", line 1799, in reloaded
    started(manager, containers, count, name)
  File \"/tmp/ansible_8msQQL/ansible_module_docker.py\", line 1776, in started
    created = manager.create_containers(delta)
  File \"/tmp/ansible_8msQQL/ansible_module_docker.py\", line 1641, in create_containers
    params['host_config'] = self.create_host_config()
  File \"/tmp/ansible_8msQQL/ansible_module_docker.py\", line 1018, in create_host_config
    return docker.utils.create_host_config(**params)
  File \"/usr/local/lib/python2.7/dist-packages/docker/utils/utils.py\", line 403, in create_host_config
    host_config['Binds'] = convert_volume_binds(binds)
  File \"/usr/local/lib/python2.7/dist-packages/docker/utils/utils.py\", line 173, in convert_volume_binds
    for k, v in binds.items():
AttributeError: 'list' object has no attribute 'items'
", "msg": "MODULE FAILURE", "parsed": false}

NO MORE HOSTS LEFT *************************************************************
 [WARNING]: Could not create retry file 'deploy.retry'.         [Errno 2] No such file or directory: ''


PLAY RECAP *********************************************************************
web01                      : ok=1    changed=0    unreachable=0    failed=1   

```

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by hoffoo** [comment]
I can confirm the same issue

```
- name: setup cadvisor
  docker:
    name: cadvisor
    image: google/cadvisor:latest
    state: started
    pull: always
    volumes:
      - /:/rootfs:ro
      - /var/run:/var/run:rw
      - /sys:/sys:ro
      - /var/lib/docker/:/var/lib/docker:ro
    ports:
      - 8080:8080
```

**Comment by ozbillwang** [comment]
yes, I got the same error in ansible v2.1.1.0, the ansible playbook runs well in ansible version 2.0.x.

I only see this error after upgrade ansible to latest. 

replace the module name from `docker` to `docker_container`, I got same error.

Rollback the version to 2.0.0.2 and 1.9.6, it runs fine.

**Comment by chouseknecht** [comment]
wontfix

The following playbook with all the same volume mounts works as expected using the latest docker_container. I'm running from source. There were commits added to docker_container on 6/1 that may not have landed in 2.1.1.0. They may be part of 2.1.1.0-1. Or, you can always run from source, or place a copy of the latest module code in a library directory. 

```
- name: Test docker_container
  hosts: localhost
  gather_facts: no
  connection: local
  tasks:
    - docker_container:
        name: cadvisor
        image: busybox:latest
        state: started
        command: sleep 1d
        pull: yes
        volumes:
          - /:/rootfs:ro
          - /var/run:/var/run:rw
          - /sys:/sys:ro
          - /var/lib/docker/:/var/lib/docker:ro
        ports:
          - 8080:8080
        debug: yes
      register: output
    - debug: var=output
```
